## Project Title:            Algerian Forest Project
### Technology:              Machine Learning Technology
### Domain:                  Agriculture
### Project Difficulty Level: Intermediate

> Problem Statement:

The Goal is to predict the temperature of the forest. This is basically a Regression problem where the forest has recorded from June 2012 to September 2012.

> Dataset to downloaded from the below link:

https://archive.ics.uci.edu/ml/datasets/Algerian+Forest+Fires+Dataset++#

In [2]:
# Importing Libraries
import numpy as numpy
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

## Display all the columns of the dataframe
pd.pandas.set_option('display.max_columns',None) # to disply all columns

### Reading Dataset

In [3]:
# ls

In [5]:
dataset = pd.read_csv("Algerian_forest_fires_dataset_UPDATE.csv")
# Finding the shape of dataset
print(f"There are {dataset.shape[0]} rows and {dataset.shape[1]} columns in the dataset\n")
dataset.head()

There are 247 rows and 1 columns in the dataset



Bejaia Region Dataset 
day month year Temperature  RH  Ws Rain  FFMC DMC DC  ISI BUI FWI              Classes  
01  06    2012 29          57  18  0     65.7 3.4 7.6 1.3 3.4 0.5            not fire   
02  06    2012 29          61  13  1.3   64.4 4.1 7.6 1   3.9 0.4            not fire   
03  06    2012 26          82  22  13.1  47.1 2.5 7.1 0.3 2.7 0.1            not fire   
04  06    2012 25          89  13  2.5   28.6 1.3 6.9 0   1.7 0              not fire

## Removing Header


In [14]:
dataset = pd.read_csv("Algerian_forest_fires_dataset_UPDATE.csv",header=1)
# Finding the shape of dataset
print(f"There are {dataset.shape[0]} rows and {dataset.shape[1]} columns in the dataset\n")
dataset.head()

There are 246 rows and 14 columns in the dataset



day month  year Temperature  RH  Ws Rain   FFMC  DMC    DC  ISI  BUI  FWI  \
0  01    06  2012          29  57  18     0  65.7  3.4   7.6  1.3  3.4  0.5   
1  02    06  2012          29  61  13   1.3  64.4  4.1   7.6    1  3.9  0.4   
2  03    06  2012          26  82  22  13.1  47.1  2.5   7.1  0.3  2.7  0.1   
3  04    06  2012          25  89  13   2.5  28.6  1.3   6.9    0  1.7    0   
4  05    06  2012          27  77  16     0  64.8    3  14.2  1.2  3.9  0.5   

     Classes    
0  not fire     
1  not fire     
2  not fire     
3  not fire     
4  not fire

In [29]:
# Finding the information about the dataset:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246 entries, 0 to 245
Data columns (total 14 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   day          246 non-null    object
 1   month        245 non-null    object
 2   year         245 non-null    object
 3   Temperature  245 non-null    object
 4    RH          245 non-null    object
 5    Ws          245 non-null    object
 6   Rain         245 non-null    object
 7   FFMC         245 non-null    object
 8   DMC          245 non-null    object
 9   DC           245 non-null    object
 10  ISI          245 non-null    object
 11  BUI          245 non-null    object
 12  FWI          245 non-null    object
 13  Classes      244 non-null    object
dtypes: object(14)
memory usage: 27.0+ KB


## need to check why datatype is object as we can see the entries are Interger.

In [33]:
dataset.day.unique()

array(['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11',
       '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22',
       '23', '24', '25', '26', '27', '28', '29', '30', '31',
       'Sidi-Bel Abbes Region Dataset', 'day'], dtype=object)

### checking the rows which has 'day' as 'Sidi-Bel Abbes Region Dataset' and  'day'

In [39]:
dataset[(dataset['day']=='Sidi-Bel Abbes Region Dataset')|(dataset['day']=='day')]

day  month  year  Temperature   RH   Ws  Rain   \
122  Sidi-Bel Abbes Region Dataset    NaN   NaN          NaN  NaN  NaN    NaN   
123                            day  month  year  Temperature   RH   Ws  Rain    

     FFMC  DMC   DC  ISI  BUI  FWI  Classes    
122   NaN  NaN  NaN  NaN  NaN  NaN        NaN  
123  FFMC  DMC   DC  ISI  BUI  FWI  Classes

In [41]:
dataset.iloc[120:].head(10)

day  month  year  Temperature   RH   Ws  Rain   \
120                             29     09  2012           26   80   16    1.8   
121                             30     09  2012           25   78   14    1.4   
122  Sidi-Bel Abbes Region Dataset    NaN   NaN          NaN  NaN  NaN    NaN   
123                            day  month  year  Temperature   RH   Ws  Rain    
124                             01     06  2012           32   71   12    0.7   
125                             02     06  2012           30   73   13      4   
126                             03     06  2012           29   80   14      2   
127                             04     06  2012           30   64   14      0   
128                             05     06  2012           32   60   14    0.2   
129                             06     06  2012           35   54   11    0.1   

     FFMC  DMC    DC  ISI  BUI  FWI    Classes    
120  47.4  2.9   7.7  0.3    3  0.1  not fire     
121    45  1.9   7.5  0.2  2.4  0.1  not fire     
122   NaN  NaN   NaN  NaN  NaN  NaN          NaN  
123  FFMC  DMC    DC  ISI  BUI  FWI    Classes    
124  57.1  2.5   8.2  0.6  2.8  0.2  not fire     
125  55.7  2.7   7.8  0.6  2.9  0.2  not fire     
126  48.7  2.2   7.6  0.3  2.6  0.1  not fire     
127  79.4  5.2  15.4  2.2  5.6    1  not fire     
128  77.1    6  17.6  1.8  6.5  0.9  not fire     
129  83.7  8.4  26.3  3.1  9.3  3.1      fire

## Observation: this dataset consists of two regions of Algerian forest.:  
- REgion 1: Bejaia Region Dataset
- REgion 2: Sidi-Bel Abbes Region Dataset
### 1. need to drop 122 and 123
### 2. need to create a categorical variable 'REgion for 'Bejaia Region Dataset' and 'Sidi-Bel Abbes Region Dataset'

In [42]:
dataset.drop(index=[122,123], inplace=True) # droping row 122,123 from dataset
dataset.iloc[120:].head(10)

day month  year Temperature  RH  Ws Rain   FFMC  DMC    DC  ISI   BUI  \
120  29    09  2012          26  80  16   1.8  47.4  2.9   7.7  0.3     3   
121  30    09  2012          25  78  14   1.4    45  1.9   7.5  0.2   2.4   
124  01    06  2012          32  71  12   0.7  57.1  2.5   8.2  0.6   2.8   
125  02    06  2012          30  73  13     4  55.7  2.7   7.8  0.6   2.9   
126  03    06  2012          29  80  14     2  48.7  2.2   7.6  0.3   2.6   
127  04    06  2012          30  64  14     0  79.4  5.2  15.4  2.2   5.6   
128  05    06  2012          32  60  14   0.2  77.1    6  17.6  1.8   6.5   
129  06    06  2012          35  54  11   0.1  83.7  8.4  26.3  3.1   9.3   
130  07    06  2012          35  44  17   0.2  85.6  9.9  28.9  5.4  10.7   
131  08    06  2012          28  51  17   1.3  71.4  7.7   7.4  1.5   7.3   

     FWI    Classes    
120  0.1  not fire     
121  0.1  not fire     
124  0.2  not fire     
125  0.2  not fire     
126  0.1  not fire     
127    1  not fire     
128  0.9  not fire     
129  3.1      fire     
130    6      fire     
131  0.8  not fire

In [43]:
# correcting the index values
dataset.reset_index(inplace=True)


dataset.iloc[121:].head()

index day month  year Temperature  RH  Ws Rain   FFMC  DMC    DC  ISI  \
121    121  30    09  2012          25  78  14   1.4    45  1.9   7.5  0.2   
122    124  01    06  2012          32  71  12   0.7  57.1  2.5   8.2  0.6   
123    125  02    06  2012          30  73  13     4  55.7  2.7   7.8  0.6   
124    126  03    06  2012          29  80  14     2  48.7  2.2   7.6  0.3   
125    127  04    06  2012          30  64  14     0  79.4  5.2  15.4  2.2   

     BUI  FWI    Classes    
121  2.4  0.1  not fire     
122  2.8  0.2  not fire     
123  2.9  0.2  not fire     
124  2.6  0.1  not fire     
125  5.6    1  not fire

In [44]:
# deleting the index feature
dataset.drop('index', axis=1, inplace=True)
dataset.iloc[120:].head(10)

day month  year Temperature  RH  Ws Rain   FFMC  DMC    DC  ISI   BUI  \
120  29    09  2012          26  80  16   1.8  47.4  2.9   7.7  0.3     3   
121  30    09  2012          25  78  14   1.4    45  1.9   7.5  0.2   2.4   
122  01    06  2012          32  71  12   0.7  57.1  2.5   8.2  0.6   2.8   
123  02    06  2012          30  73  13     4  55.7  2.7   7.8  0.6   2.9   
124  03    06  2012          29  80  14     2  48.7  2.2   7.6  0.3   2.6   
125  04    06  2012          30  64  14     0  79.4  5.2  15.4  2.2   5.6   
126  05    06  2012          32  60  14   0.2  77.1    6  17.6  1.8   6.5   
127  06    06  2012          35  54  11   0.1  83.7  8.4  26.3  3.1   9.3   
128  07    06  2012          35  44  17   0.2  85.6  9.9  28.9  5.4  10.7   
129  08    06  2012          28  51  17   1.3  71.4  7.7   7.4  1.5   7.3   

     FWI    Classes    
120  0.1  not fire     
121  0.1  not fire     
122  0.2  not fire     
123  0.2  not fire     
124  0.1  not fire     
125    1  not fire     
126  0.9  not fire     
127  3.1      fire     
128    6      fire     
129  0.8  not fire

### from index 122 the regions is 'Sidi-Bel Abbes Region Dataset'

In [46]:
### creating feature called Region 0 for Bejaia region and 1 for Sidi Bel-abbes region
dataset.loc[:122,'Region']=0
dataset.loc[122:, 'Region']=1

dataset.iloc[120:].head(10)

day month  year Temperature  RH  Ws Rain   FFMC  DMC    DC  ISI   BUI  \
120  29    09  2012          26  80  16   1.8  47.4  2.9   7.7  0.3     3   
121  30    09  2012          25  78  14   1.4    45  1.9   7.5  0.2   2.4   
122  01    06  2012          32  71  12   0.7  57.1  2.5   8.2  0.6   2.8   
123  02    06  2012          30  73  13     4  55.7  2.7   7.8  0.6   2.9   
124  03    06  2012          29  80  14     2  48.7  2.2   7.6  0.3   2.6   
125  04    06  2012          30  64  14     0  79.4  5.2  15.4  2.2   5.6   
126  05    06  2012          32  60  14   0.2  77.1    6  17.6  1.8   6.5   
127  06    06  2012          35  54  11   0.1  83.7  8.4  26.3  3.1   9.3   
128  07    06  2012          35  44  17   0.2  85.6  9.9  28.9  5.4  10.7   
129  08    06  2012          28  51  17   1.3  71.4  7.7   7.4  1.5   7.3   

     FWI    Classes    Region  
120  0.1  not fire        0.0  
121  0.1  not fire        0.0  
122  0.2  not fire        1.0  
123  0.2  not fire        1.0  
124  0.1  not fire        1.0  
125    1  not fire        1.0  
126  0.9  not fire        1.0  
127  3.1      fire        1.0  
128    6      fire        1.0  
129  0.8  not fire        1.0

## Checking data info Again:


In [47]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   day          244 non-null    object 
 1   month        244 non-null    object 
 2   year         244 non-null    object 
 3   Temperature  244 non-null    object 
 4    RH          244 non-null    object 
 5    Ws          244 non-null    object 
 6   Rain         244 non-null    object 
 7   FFMC         244 non-null    object 
 8   DMC          244 non-null    object 
 9   DC           244 non-null    object 
 10  ISI          244 non-null    object 
 11  BUI          244 non-null    object 
 12  FWI          244 non-null    object 
 13  Classes      243 non-null    object 
 14  Region       244 non-null    float64
dtypes: float64(1), object(14)
memory usage: 28.7+ KB


In [49]:
unique_feature=[feature for feature in dataset.columns if len(dataset[feature].unique())==len(dataset)]
print("Unique feature Count {} out of {} features".format(len(unique_feature),len(dataset.columns)))
print(unique_feature)
# seperating the Numerical and Categorical Variables:
numerical_features = [fea for fea in dataset.columns if dataset[fea].dtype != 'O']
print("Numerical feature Count {}".format(len(numerical_features)))
print(numerical_features)
unique_feature=[feature for feature in numerical_features if len(dataset[feature].unique())==len(dataset)]
print("Unique feature Count {} out of {} features".format(len(unique_feature),len(numerical_features)))
## Numerical variables are usually of 2 type
## 1. Continous variable and Discrete Variables
discrete_feature=[feature for feature in numerical_features if len(dataset[feature].unique())<25 and feature not in unique_feature]
continuous_feature=[feature for feature in numerical_features if feature not in discrete_feature+unique_feature]
print("Continuous feature Count {} out of {} features".format(len(continuous_feature),len(numerical_features)))
print(continuous_feature)
print("Discrete Variables Count: {} out of {} features".format(len(discrete_feature),len(numerical_features)))
print(discrete_feature)
data = dataset.copy()
pd.DataFrame(list(zip(discrete_feature,[len(data[feature].unique()) for feature in discrete_feature], [data[str(feature)].unique() for feature in discrete_feature])), columns=['Discrete Features','No of Unique Values', 'Unique Values'])



Unique feature Count 0 out of 15 features
[]
Numerical feature Count 1
['Region']
Unique feature Count 0 out of 1 features
Continuous feature Count 0 out of 1 features
[]
Discrete Variables Count: 1 out of 1 features
['Region']


Discrete Features  No of Unique Values Unique Values
0            Region                    2    [0.0, 1.0]

In [50]:
categorical_features = [fea for fea in dataset.columns if dataset[fea].dtype == 'O']
print("Categorical feature Count {}".format(len(categorical_features)))
print(categorical_features)
data = dataset.copy()
pd.DataFrame(list(zip(categorical_features,[len(data[feature].unique()) for feature in categorical_features], [data[str(feature)].unique() for feature in categorical_features])), columns=['Categorical Features','No. of Categories', 'Categories'])

Categorical feature Count 14
['day', 'month', 'year', 'Temperature', ' RH', ' Ws', 'Rain ', 'FFMC', 'DMC', 'DC', 'ISI', 'BUI', 'FWI', 'Classes  ']


Categorical Features  No. of Categories  \
0                   day                 31   
1                 month                  4   
2                  year                  1   
3           Temperature                 19   
4                    RH                 62   
5                    Ws                 18   
6                 Rain                  39   
7                  FFMC                173   
8                   DMC                166   
9                    DC                198   
10                  ISI                106   
11                  BUI                174   
12                  FWI                127   
13            Classes                    9   

                                           Categories  
0   [01, 02, 03, 04, 05, 06, 07, 08, 09, 10, 11, 1...  
1                                    [06, 07, 08, 09]  
2                                              [2012]  
3   [29, 26, 25, 27, 31, 33, 30, 28, 32, 34, 35, 3...  
4   [57, 61, 82, 89, 77, 67, 54, 73, 88, 79, 65, 8...  
5   [18, 13, 22, 16, 14, 15, 12, 19, 21, 20, 17, 2...  
6   [0, 1.3, 13.1, 2.5, 0.2, 1.2, 0.5, 3.1, 0.7, 0...  
7   [65.7, 64.4, 47.1, 28.6, 64.8, 82.6, 88.2, 86....  
8   [3.4, 4.1, 2.5, 1.3, 3, 5.8, 9.9, 12.1, 7.9, 9...  
9   [7.6, 7.1, 6.9, 14.2, 22.2, 30.5, 38.3, 38.8, ...  
10  [1.3, 1, 0.3, 0, 1.2, 3.1, 6.4, 5.6, 0.4, 4, 4...  
11  [3.4, 3.9, 2.7, 1.7, 7, 10.9, 13.5, 10.5, 12.6...  
12  [0.5, 0.4, 0.1, 0, 2.5, 7.2, 7.1, 0.3, 0.9, 5....  
13  [not fire   , fire   , fire, fire , not fire, ...

In [51]:
[data[str(feature)].unique() for feature in categorical_features]


[array(['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11',
        '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22',
        '23', '24', '25', '26', '27', '28', '29', '30', '31'], dtype=object),
 array(['06', '07', '08', '09'], dtype=object),
 array(['2012'], dtype=object),
 array(['29', '26', '25', '27', '31', '33', '30', '28', '32', '34', '35',
        '36', '37', '22', '24', '38', '39', '40', '42'], dtype=object),
 array(['57', '61', '82', '89', '77', '67', '54', '73', '88', '79', '65',
        '81', '84', '78', '80', '55', '62', '66', '64', '53', '47', '50',
        '68', '75', '76', '63', '69', '70', '59', '48', '45', '60', '51',
        '52', '58', '86', '74', '71', '49', '44', '41', '42', '90', '87',
        '72', '46', '37', '36', '56', '43', '83', '29', '34', '33', '35',
        '39', '31', '21', '40', '24', '38', '26'], dtype=object),
 array(['18', '13', '22', '16', '14', '15', '12', '19', '21', '20', '17',
        '26', '11', '10', '9', '8', '6

# Observation:

- Need to handle the spaces in Column name
- Need to Handle the spaces in values in features
- Need to handle the 'fire' entry in Column name 'FWI'
- Need to Handle the Datatype.

In [53]:
dataset.columns

Index(['day', 'month', 'year', 'Temperature', ' RH', ' Ws', 'Rain ', 'FFMC',
       'DMC', 'DC', 'ISI', 'BUI', 'FWI', 'Classes  ', 'Region'],
      dtype='object')

In [54]:
# stripping spaces from column names
dataset.columns= [col_name.strip() for col_name in dataset.columns]
dataset.columns

Index(['day', 'month', 'year', 'Temperature', 'RH', 'Ws', 'Rain', 'FFMC',
       'DMC', 'DC', 'ISI', 'BUI', 'FWI', 'Classes', 'Region'],
      dtype='object')

In [55]:
### converting all feature values to string so that we can do data cleaning as shown below.
dataset=dataset.astype(str)
### somes values in colums also have space
for feature in dataset.columns:
    dataset[feature]= dataset[feature].str.replace(" ","")

In [56]:
### index no 165 for feature name FWI has value fire
dataset[dataset['FWI']== 'fire'].index

Int64Index([165], dtype='int64')

In [60]:
### replacing fire value witha float value
dataset.drop(165,inplace=True) 
dataset.iloc[160:].head(10)

day month  year Temperature  RH  Ws Rain  FFMC   DMC    DC   ISI   BUI  \
160  09    07  2012          36  43  15  1.9  82.3   9.4   9.9   3.2     9   
161  10    07  2012          34  51  16  3.8  77.5     8   9.5     2   7.7   
162  11    07  2012          34  56  15  2.9  74.8   7.1   9.5   1.6   6.8   
163  12    07  2012          36  44  13    0  90.1  12.6  19.4   8.3  12.5   
164  13    07  2012          39  45  13  0.6  85.2  11.3  10.4   4.2  10.9   
166  15    07  2012          34  45  17    0  90.5    18  24.1  10.9  17.7   
167  16    07  2012          31  83  17    0  84.5  19.4  33.1   4.7  19.2   
168  17    07  2012          32  81  17    0  84.6  21.1  42.3   4.7  20.9   
169  18    07  2012          33  68  15    0  86.1  23.9  51.6   5.2  23.9   
170  19    07  2012          34  58  16    0  88.1  27.8  61.1   7.3  27.7   

      FWI  Classes Region  
160   3.1     fire    1.0  
161   1.3  notfire    1.0  
162   0.8  notfire    1.0  
163   9.6     fire    1.0  
164   4.7     fire    1.0  
166  14.1     fire    1.0  
167   7.3     fire    1.0  
168   7.7     fire    1.0  
169   9.1     fire    1.0  
170    13     fire    1.0

In [61]:
# correcting the index values
dataset.reset_index(inplace=True)
dataset.iloc[160:].head(10)

index day month  year Temperature  RH  Ws Rain  FFMC   DMC    DC   ISI  \
160    160  09    07  2012          36  43  15  1.9  82.3   9.4   9.9   3.2   
161    161  10    07  2012          34  51  16  3.8  77.5     8   9.5     2   
162    162  11    07  2012          34  56  15  2.9  74.8   7.1   9.5   1.6   
163    163  12    07  2012          36  44  13    0  90.1  12.6  19.4   8.3   
164    164  13    07  2012          39  45  13  0.6  85.2  11.3  10.4   4.2   
165    166  15    07  2012          34  45  17    0  90.5    18  24.1  10.9   
166    167  16    07  2012          31  83  17    0  84.5  19.4  33.1   4.7   
167    168  17    07  2012          32  81  17    0  84.6  21.1  42.3   4.7   
168    169  18    07  2012          33  68  15    0  86.1  23.9  51.6   5.2   
169    170  19    07  2012          34  58  16    0  88.1  27.8  61.1   7.3   

      BUI   FWI  Classes Region  
160     9   3.1     fire    1.0  
161   7.7   1.3  notfire    1.0  
162   6.8   0.8  notfire    1.0  
163  12.5   9.6     fire    1.0  
164  10.9   4.7     fire    1.0  
165  17.7  14.1     fire    1.0  
166  19.2   7.3     fire    1.0  
167  20.9   7.7     fire    1.0  
168  23.9   9.1     fire    1.0  
169  27.7    13     fire    1.0

In [62]:
# deleting the index feature
dataset.drop('index', axis=1, inplace=True)
dataset.iloc[160:].head(10)

day month  year Temperature  RH  Ws Rain  FFMC   DMC    DC   ISI   BUI  \
160  09    07  2012          36  43  15  1.9  82.3   9.4   9.9   3.2     9   
161  10    07  2012          34  51  16  3.8  77.5     8   9.5     2   7.7   
162  11    07  2012          34  56  15  2.9  74.8   7.1   9.5   1.6   6.8   
163  12    07  2012          36  44  13    0  90.1  12.6  19.4   8.3  12.5   
164  13    07  2012          39  45  13  0.6  85.2  11.3  10.4   4.2  10.9   
165  15    07  2012          34  45  17    0  90.5    18  24.1  10.9  17.7   
166  16    07  2012          31  83  17    0  84.5  19.4  33.1   4.7  19.2   
167  17    07  2012          32  81  17    0  84.6  21.1  42.3   4.7  20.9   
168  18    07  2012          33  68  15    0  86.1  23.9  51.6   5.2  23.9   
169  19    07  2012          34  58  16    0  88.1  27.8  61.1   7.3  27.7   

      FWI  Classes Region  
160   3.1     fire    1.0  
161   1.3  notfire    1.0  
162   0.8  notfire    1.0  
163   9.6     fire    1.0  
164   4.7     fire    1.0  
165  14.1     fire    1.0  
166   7.3     fire    1.0  
167   7.7     fire    1.0  
168   9.1     fire    1.0  
169    13     fire    1.0

In [63]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 243 entries, 0 to 242
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   day          243 non-null    object
 1   month        243 non-null    object
 2   year         243 non-null    object
 3   Temperature  243 non-null    object
 4   RH           243 non-null    object
 5   Ws           243 non-null    object
 6   Rain         243 non-null    object
 7   FFMC         243 non-null    object
 8   DMC          243 non-null    object
 9   DC           243 non-null    object
 10  ISI          243 non-null    object
 11  BUI          243 non-null    object
 12  FWI          243 non-null    object
 13  Classes      243 non-null    object
 14  Region       243 non-null    object
dtypes: object(15)
memory usage: 28.6+ KB


In [65]:
### changing datatypes of features to numerical for numerical features as all are in object

datatype_convert={'day':'int64','month':'int64','year':'int64','Temperature':'int64','RH':'int64', 'Ws':'int64','Rain':'float64',
                  'FFMC':'float64', 'DMC':'float64', 'DC':'float64', 'ISI':'float64', 'BUI':'float64', 'FWI':'float64', 
                 'Region':'float64'}

dataset=dataset.astype(datatype_convert)
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 243 entries, 0 to 242
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   day          243 non-null    int64  
 1   month        243 non-null    int64  
 2   year         243 non-null    int64  
 3   Temperature  243 non-null    int64  
 4   RH           243 non-null    int64  
 5   Ws           243 non-null    int64  
 6   Rain         243 non-null    float64
 7   FFMC         243 non-null    float64
 8   DMC          243 non-null    float64
 9   DC           243 non-null    float64
 10  ISI          243 non-null    float64
 11  BUI          243 non-null    float64
 12  FWI          243 non-null    float64
 13  Classes      243 non-null    object 
 14  Region       243 non-null    float64
dtypes: float64(8), int64(6), object(1)
memory usage: 28.6+ KB


# Data Profile

In [73]:
dataset.head()

day  month  year  Temperature  RH  Ws  Rain  FFMC  DMC    DC  ISI  BUI  \
0    1      6  2012           29  57  18   0.0  65.7  3.4   7.6  1.3  3.4   
1    2      6  2012           29  61  13   1.3  64.4  4.1   7.6  1.0  3.9   
2    3      6  2012           26  82  22  13.1  47.1  2.5   7.1  0.3  2.7   
3    4      6  2012           25  89  13   2.5  28.6  1.3   6.9  0.0  1.7   
4    5      6  2012           27  77  16   0.0  64.8  3.0  14.2  1.2  3.9   

   FWI  Classes  Region  
0  0.5  notfire     0.0  
1  0.4  notfire     0.0  
2  0.1  notfire     0.0  
3  0.0  notfire     0.0  
4  0.5  notfire     0.0

In [70]:
print("Total Features Count is {}: {}".format(len(dataset.columns),dataset.columns))

unique_feature=[feature for feature in dataset.columns if len(dataset[feature].unique())==len(dataset)]
print("\n Unique feature Count {} out of {} features".format(len(unique_feature),len(dataset.columns)))
print(unique_feature)
# seperating the Numerical and Categorical Variables:
numerical_features = [fea for fea in dataset.columns if dataset[fea].dtype != 'O']
print("\n Numerical feature Count {}".format(len(numerical_features)))
print(numerical_features)
unique_feature=[feature for feature in numerical_features if len(dataset[feature].unique())==len(dataset)]
print("\n Unique feature Count {} out of {} features".format(len(unique_feature),len(numerical_features)))
## Numerical variables are usually of 2 type
## 1. Continous variable and Discrete Variables
discrete_feature=[feature for feature in numerical_features if len(dataset[feature].unique())<25 and feature not in unique_feature]
continuous_feature=[feature for feature in numerical_features if feature not in discrete_feature+unique_feature]
print("\n Continuous feature Count {} out of {} features".format(len(continuous_feature),len(numerical_features)))
print(continuous_feature)
print("\n Discrete Variables Count: {} out of {} features".format(len(discrete_feature),len(numerical_features)))
print(discrete_feature)
data = dataset.copy()
pd.DataFrame(list(zip(discrete_feature,[len(data[feature].unique()) for feature in discrete_feature], [data[str(feature)].unique() for feature in discrete_feature])), columns=['Discrete Features','No of Unique Values', 'Unique Values'])



Total Features Count is 15: Index(['day', 'month', 'year', 'Temperature', 'RH', 'Ws', 'Rain', 'FFMC',
       'DMC', 'DC', 'ISI', 'BUI', 'FWI', 'Classes', 'Region'],
      dtype='object')

 Unique feature Count 0 out of 15 features
[]

 Numerical feature Count 14
['day', 'month', 'year', 'Temperature', 'RH', 'Ws', 'Rain', 'FFMC', 'DMC', 'DC', 'ISI', 'BUI', 'FWI', 'Region']

 Unique feature Count 0 out of 14 features

 Continuous feature Count 9 out of 14 features
['day', 'RH', 'Rain', 'FFMC', 'DMC', 'DC', 'ISI', 'BUI', 'FWI']

 Discrete Variables Count: 5 out of 14 features
['month', 'year', 'Temperature', 'Ws', 'Region']


Discrete Features  No of Unique Values  \
0             month                    4   
1              year                    1   
2       Temperature                   19   
3                Ws                   18   
4            Region                    2   

                                       Unique Values  
0                                       [6, 7, 8, 9]  
1                                             [2012]  
2  [29, 26, 25, 27, 31, 33, 30, 28, 32, 34, 35, 3...  
3  [18, 13, 22, 16, 14, 15, 12, 19, 21, 20, 17, 2...  
4                                         [0.0, 1.0]

In [67]:
categorical_features = [fea for fea in dataset.columns if dataset[fea].dtype == 'O']
print("Categorical feature Count {}".format(len(categorical_features)))
print(categorical_features)
data = dataset.copy()
pd.DataFrame(list(zip(categorical_features,[len(data[feature].unique()) for feature in categorical_features], [data[str(feature)].unique() for feature in categorical_features])), columns=['Categorical Features','No. of Categories', 'Categories'])

Categorical feature Count 1
['Classes']


Categorical Features  No. of Categories       Categories
0              Classes                  2  [notfire, fire]

In [68]:
[data[str(feature)].unique() for feature in categorical_features]


[array(['notfire', 'fire'], dtype=object)]

In [72]:
# Checking Null values in dataset
dataset.isnull().sum().sort_values(ascending=False)

day            0
month          0
year           0
Temperature    0
RH             0
Ws             0
Rain           0
FFMC           0
DMC            0
DC             0
ISI            0
BUI            0
FWI            0
Classes        0
Region         0
dtype: int64

In [76]:
dataset.describe(include='all').T

count unique   top freq       mean        std     min     25%  \
day          243.0    NaN   NaN  NaN  15.761317   8.842552     1.0     8.0   
month        243.0    NaN   NaN  NaN   7.502058   1.114793     6.0     7.0   
year         243.0    NaN   NaN  NaN     2012.0        0.0  2012.0  2012.0   
Temperature  243.0    NaN   NaN  NaN  32.152263   3.628039    22.0    30.0   
RH           243.0    NaN   NaN  NaN  62.041152   14.82816    21.0    52.5   
Ws           243.0    NaN   NaN  NaN  15.493827   2.811385     6.0    14.0   
Rain         243.0    NaN   NaN  NaN   0.762963   2.003207     0.0     0.0   
FFMC         243.0    NaN   NaN  NaN  77.842387  14.349641    28.6   71.85   
DMC          243.0    NaN   NaN  NaN  14.680658   12.39304     0.7     5.8   
DC           243.0    NaN   NaN  NaN  49.430864  47.665606     6.9   12.35   
ISI          243.0    NaN   NaN  NaN   4.742387   4.154234     0.0     1.4   
BUI          243.0    NaN   NaN  NaN  16.690535  14.228421     1.1     6.0   
FWI          243.0    NaN   NaN  NaN   7.035391   7.440568     0.0     0.7   
Classes        243      2  fire  137        NaN        NaN     NaN     NaN   
Region       243.0    NaN   NaN  NaN   0.497942   0.501028     0.0     0.0   

                50%     75%     max  
day            16.0    23.0    31.0  
month           8.0     8.0     9.0  
year         2012.0  2012.0  2012.0  
Temperature    32.0    35.0    42.0  
RH             63.0    73.5    90.0  
Ws             15.0    17.0    29.0  
Rain            0.0     0.5    16.8  
FFMC           83.3    88.3    96.0  
DMC            11.3    20.8    65.9  
DC             33.1    69.1   220.4  
ISI             3.5    7.25    19.0  
BUI            12.4   22.65    68.0  
FWI             4.2   11.45    31.1  
Classes         NaN     NaN     NaN  
Region          0.0     1.0     1.0

### 1.6 Info about dataset and its attributes

1. The dataset includes 243 instances that regroup a data of two regions of Algeria,namely the Bejaia region located in the northeast of Algeria and the Sidi Bel-abbes region located in the northwest of Algeria.
2. 122 instances for Bejaia region and 121 for Sidi Bel-abbes region.
3. The period from June 2012 to September 2012.
4. The dataset includes 11 attribues and 1 output attribue (class)
5. The 243 instances have been classified into fire (137 classes) and notfire (106 classes) classes.

### Attributes

1. Date : (DD/MM/YYYY) Day, month ('june' to 'september'), year (2012)

**Weather data observations**

2. Temp : temperature noon (temperature max) in Celsius degrees: 22 to 42
3. RH : Relative Humidity in %: 21 to 90
4. Ws :Wind speed in km/h: 6 to 29
5. Rain: total day in mm: 0 to 16.8

**FWI Components**

6. Fine Fuel Moisture Code (FFMC) index from the FWI system: 28.6 to 92.5
7. Duff Moisture Code (DMC) index from the FWI system: 1.1 to 65.9
8. Drought Code (DC) index from the FWI system: 7 to 220.4
9. Initial Spread Index (ISI) index from the FWI system: 0 to 18.5
10. Buildup Index (BUI) index from the FWI system: 1.1 to 68
11. Fire Weather Index (FWI) Index: 0 to 31.1
12. Classes: two classes, namely fire and not fire

In [77]:
from sklearn.model_selection import train_test_split
train, test = train_test_split(dataset, test_size = 1/3, random_state = 0)

In [78]:
# Finding the information about the dataset:
train.shape,  test.shape

((162, 15), (81, 15))

In [79]:
# Saving dateset train and test 
train.to_csv('train.csv',index=False)
# Saving dateset with cleaned data
test.to_csv('test.csv',index=False)